### PR TITLE
update for supportserver image creation and usage

### PR DESCRIPTION
**Pull request by psladek**

### CODE REVIEW

File: main.pm [hunk 1]
```diff
@@ -99,6 +99,10 @@ if (get_var("LIVETEST")) {
     $testapi::username = "root";
     $testapi::password = '';
 }
+elsif (get_var("AUTOYAST") || get_var("SUPPORT_SERVER") ) {
+    $testapi::username = "root";
+    $testapi::password = "root";
+}
```
Comment (sysrich) [inline main.pm]: There are autoyast tests besides the ones you're writing. The autoyast tests being put together by QA use bernhard/nots3cr3t like our usual tests, so I really dont like this

Users shouldn't use root unless they need it, so our tests shouldn't use root unless they need it

File: main.pm [hunk 2]
```diff
@@ -562,6 +566,20 @@ sub load_x11tests(){
...
 }
 
+sub load_autoyast_tests(){
+#    init boot in load_boot_tests
+    loadtest("autoyast/system.pm");
+    loadtest("autoyast/console.pm");
+    loadtest("autoyast/login.pm");
+    loadtest("autoyast/repos.pm") unless get_var("SUPPORT_SERVER_GENERATOR");
+    loadtest("autoyast/autoyast_verify.pm") if get_var("AUTOYAST_VERIFY");
+    loadtest("autoyast/useradd.pm") unless get_var("INSTALLONLY");
```
Comment (sysrich) [inline main.pm]: Why do we need a test to add a user? why can't we have autoyast templates that do that properly?

Users aren't going to have magical openQA to add a user after their installation using autoyast, so we really need to be testing the addition of users VIA autoyast..not scripting around it with openQA tests

File: main.pm (hunk 1 above)
Comment (psladek) [inline main.pm]: yes, the autoyast tests in tests/autoyast needs root access. I guess there are now some functions to log in as root, so this redefinition may no longer be needed, and the root password can already be specified in autoyast.xml, as nots3cr3t
I just thought this is preferred way to do this, according to the lines above.

File: main.pm (hunk 2 above)
Comment (psladek) [inline main.pm]: This was done to minimize modifications done to autoyast files from e.g. bugreports, but this seems not so neccessary.
Yes ok, as in the above comment, there is no problem with user to be added already into autoyast.xml.

**Comment by lnussel** [comment]
I'll merge it to get it off the list. Please consider addressing Richard's concern nevertheless :)

**Pull request merged**